controllers/virtualmachinesnapshot/virtualmachinesnapshot_controller_intg_test.go

```diff
@@ -469,18 +437,8 @@ func intgTestsReconcile() {
 							g.Expect(vmObj.Status.CurrentSnapshot.Name).To(Equal(vmSnapshotL1.Name))
 						}).Should(Succeed(), "waiting for current snapshot to be updated to root")
 
-						By("check parent snapshot's children should be updated")
-						parent := vmSnapshotL1
-						parentSSObjKey := types.NamespacedName{Name: parent.Name, Namespace: parent.Namespace}
-						Eventually(func(g Gomega) {
-							tmpParent := &vmopv1.VirtualMachineSnapshot{}
-							g.Expect(vcSimCtx.Client.Get(ctx, parentSSObjKey, tmpParent)).To(Succeed())
-							g.Expect(tmpParent).To(Not(BeNil()))
-							g.Expect(tmpParent.Status.Children).To(HaveLen(2))
-							g.Expect(tmpParent.Status.Children).To(ContainElement(*newLocalObjectRefWithSnapshotName(vmSnapshotL3Node1Name)))
-							g.Expect(tmpParent.Status.Children).To(ContainElement(*newLocalObjectRefWithSnapshotName(vmSnapshotL3Node2Name)))
-							g.Expect(tmpParent.Status.Children).ToNot(ContainElement(*newLocalObjectRefWithSnapshotName(vmSnapshotL2Name)))
-						}).Should(Succeed(), "waiting for parent snapshot's children to be updated")
+						// Skip checking parent snapshot's children because it is synced through fakeVMProvider.SyncVMSnapshotTreeStatusFn
```

Comment:
What do you mean? You can set the Fn to whatever capture variables, no?

Reply:
I guess I could try add k8sClient as a func to fakeVMProvider, and see if that works. Let me try

Reply:
NVM, I can call the vcSimCtx inside a Fn .. Didn't know I can do that